pkg/operator/ceph/object/notification/obc_label_controller.go

```diff
@@ -136,11 +136,14 @@ func (r *ReconcileOBCLabels) reconcile(request reconcile.Request) (reconcile.Res
 		owner:            ob.Spec.AdditionalState[bucket.CephUser],
 		objectStoreName:  objectStoreName,
 	}
-	err = deleteAllNotificationsFunc(p, &ob)
+	bnList, err := getAllNotificationsFunc(p, &ob)
 	if err != nil {
 		return reconcile.Result{}, errors.Wrapf(err, "failed delete all bucket notifications from ObjectbucketClaim %q", bucketName)
 	}
 
+	labelList := make([]string, 0)
+	addList := make([]string, 0)
+	deleteList := make([]string, 0)
 	// looking for notifications in the labels
 	for labelKey, labelValue := range obc.Labels {
```

Comment:
This `reconcile` function is pretty huge, and this for loop is now a large part of it. What if we made this for loop that extracts the labels its own function and unit tested it?

Reply:
The `creating notification loop` is a large one with a lot of checks, I will move it as a separate function. Others are small loops(I agree that there are a lot of loops)